tasks/discover-granules/index.js

```diff
@@ -24,11 +24,13 @@ const logger = () => new Logger({
  *
  * @param {Object} providerConfig - the connection config for the provider
  * @param {bool} useList - flag to tell ftp server to use 'LIST' instead of 'STAT'
+ * @param {number} httpRequestTimeout - seconds for http provider to wait
+ *                                      before timing out
  * @param {string} path - the provider path to search
  * @returns {Array<Object>} a list of discovered file objects
  */
-const listFiles = (providerConfig, useList, path) =>
-  buildProviderClient({ ...providerConfig, useList }).list(path);
+const listFiles = (providerConfig, useList, httpRequestTimeout = 300, path) =>
```

Comment:
Consider updating this to take named parameters rather than positional.

Reply:
583270099

Reply:
583270099